101-vm-user-image-data-disks/azuredeploy.json

```diff
@@ -77,20 +77,31 @@
     "publicIPAddressName": "[concat(parameters('vmName'),'PublicIP')]",
     "vmName": "[parameters('vmName')]",
     "virtualNetworkName": "[concat(parameters('vmName'),'VNet')]",
-    "nicName": "[concat(parameters('vmName'),'Nic')]",
+    "nicName": "[concat(parameters('vmName'),'-nic1')]",
     "addressPrefix": "10.0.0.0/16",
     "subnet1Name": "Subnet-1",
     "subnet1Prefix": "10.0.0.0/24",
     "publicIPAddressType": "Dynamic",
     "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
     "subnet1Ref": "[concat(variables('vnetID'),'/subnets/',variables('subnet1Name'))]",
     "osDiskUrl": "[parameters('osDiskUrlParam')]",
-    "dataDiskUrl": "[parameters('dataDiskUrlParam')]",    
-    "osDiskVhdName": "[concat(variables('osDiskUrl'),'-',variables('vmName'),'.vhd')]",
-    "dataDiskVhdName": "[concat(variables('dataDiskUrl'),'-',variables('vmName'),'.vhd')]"
+    "dataDiskUrl": "[parameters('dataDiskUrlParam')]",
+    "storageAccountName": "[split( split( parameters('osDiskUrlParam'), '/')[2], '.' )[0]]",
```

Comment:
I think you need a take() here, to make sure you have a single array element in the var (i.e. end up with a string not an array)